cartographer/mapping/map_builder.cc

```diff
@@ -201,17 +201,20 @@ void MapBuilder::LoadMap(io::ProtoStreamReader* const reader) {
   // TODO(whess): Not all trajectories should be builders, i.e. support should
   // be added for trajectories without latest pose, options, etc. Appease the
   // trajectory builder for now.
-  proto::TrajectoryBuilderOptions unused_options;
-  unused_options.mutable_trajectory_builder_2d_options()
-      ->mutable_submaps_options()
-      ->set_resolution(0.05);
-  unused_options.mutable_trajectory_builder_3d_options();
-
-  const std::unordered_set<string> unused_sensor_ids;
-  const int map_trajectory_id =
-      AddTrajectoryBuilder(unused_sensor_ids, unused_options);
-  FinishTrajectory(map_trajectory_id);
-  sparse_pose_graph_->FreezeTrajectory(map_trajectory_id);
+  for (int trajectory_id = 0; trajectory_id < pose_graph.trajectory().size();
```

Comment:
I'm wondering whether this should not be implemented as a separate function `DeserializeState()`. While this is adding support for deserializing more of the serialized state, it also no longer works with trimmed data. Before this was supported.

Reply:
I planned that when `LoadMap` learns how to load an unfrozen trajectory, it should be renamed to `LoadState`. That happens in #554. I wanted to preserve all existing functionality in `LoadState` by calling it with `load_frozen_trajectories = true`.